clientv3/integration/cluster_test.go

```diff
@@ -186,4 +207,18 @@ func TestMemberAddForLearner(t *testing.T) {
 	if len(learners) != 1 {
 		t.Errorf("Added 1 learner node to cluster, got %d", len(learners))
 	}
+	_, err = capi.MemberPromote(context.Background(), resp.Member.ID)
+
+	if err != nil {
+		t.Fatalf("failed to promote member error: %v", err)
+	}
+
+	learners, err = clus.GetLearnerMembers()
```

Comment:
https://travis-ci.com/etcd-io/etcd/jobs/193852177#L1193

Found a race. 
Getting learner members is not linearizable read. `GetLearnerMembers` reads members from `clus.client(0)`, whereas promote is requested via `clus.Randclient()`. So `GetLearnerMembers` could have stale information.

Maybe add 1 second delay to mitigate the bug for now.

Reply:
Another way to mitigate, get learner members from the same client that was used to request promote.